### ISSUE TITLE
Flaky test - prometheusreceiver - TestStalenessMarkersEndToEnd

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-collector-contrib#6243]
Seen on a local run:

```
=== RUN   TestStalenessMarkersEndToEnd
==================
WARNING: DATA RACE
Read at 0x000005e8b254 by goroutine 109:
  go.opentelemetry.io/collector/obsreport.(*Receiver).endOp()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/obsreport/obsreport_receiver.go:164 +0xa4
  go.opentelemetry.io/collector/obsreport.(*Receiver).EndMetricsOp()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/obsreport/obsreport_receiver.go:119 +0x627
  github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal.(*transaction).Commit()
      /home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/transaction.go:208 +0x4c5
  github.com/prometheus/prometheus/scrape.(*timeLimitAppender).Commit()
      <autogenerated>:1 +0x4e
  github.com/prometheus/prometheus/scrape.(*scrapeLoop).scrapeAndReport.func1()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:1195 +0x99
  github.com/prometheus/prometheus/scrape.(*scrapeLoop).scrapeAndReport()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:1262 +0x1822
  github.com/prometheus/prometheus/scrape.(*scrapeLoop).run()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:1148 +0x4d5
  github.com/prometheus/prometheus/scrape.(*scrapePool).sync·dwrap·18()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:564 +0x75

Previous write at 0x000005e8b254 by goroutine 97:
  go.opentelemetry.io/collector/internal/obsreportconfig.Configure()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/internal/obsreportconfig/obsreportconfig.go:38 +0x31
  go.opentelemetry.io/collector/service.(*colTelemetry).initOpenCensus()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/service/telemetry.go:149 +0x9a
  go.opentelemetry.io/collector/service.(*colTelemetry).initOnce()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/service/telemetry.go:109 +0x431
  go.opentelemetry.io/collector/service.(*colTelemetry).init.func1()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/service/telemetry.go:70 +0x67
  sync.(*Once).doSlow()
      /usr/local/lib/go/src/sync/once.go:68 +0x127
  sync.(*Once).Do()
      /usr/local/lib/go/src/sync/once.go:59 +0x46
  go.opentelemetry.io/collector/service.(*colTelemetry).init()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/service/telemetry.go:68 +0xd1
  go.opentelemetry.io/collector/service.(*Collector).Run()
      /home/jpkroehling/go/pkg/mod/go.opentelemetry.io/collector@v0.38.1-0.20211112010215-bcb90636ae66/service/collector.go:231 +0x4a3
  github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal_test.TestStalenessMarkersEndToEnd.func4()
      /home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/staleness_end_to_end_test.go:175 +0x64

Goroutine 109 (running) created at:
  github.com/prometheus/prometheus/scrape.(*scrapePool).sync()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:564 +0xfe4
  github.com/prometheus/prometheus/scrape.(*scrapePool).Sync()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/scrape.go:480 +0x384
  github.com/prometheus/prometheus/scrape.(*Manager).reload.func1()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/manager.go:197 +0x53
  github.com/prometheus/prometheus/scrape.(*Manager).reload·dwrap·3()
      /home/jpkroehling/go/pkg/mod/github.com/prometheus/prometheus@v1.8.2-0.20210621150501-ff58416a0b02/scrape/manager.go:199 +0x74

Goroutine 97 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal_test.TestStalenessMarkersEndToEnd()
      /home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/staleness_end_to_end_test.go:174 +0xd24
  testing.tRunner()
      /usr/local/lib/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/lib/go/src/testing/testing.go:1306 +0x47
==================
    testing.go:1152: race detected during execution of test
--- FAIL: TestStalenessMarkersEndToEnd (60.03s)
=== CONT  
    testing.go:1152: race detected during execution of test
FAIL
coverage: 84.5% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal	60.196s
FAIL
make[2]: *** [../../Makefile.Common:50: do-unit-tests-with-cover] Error 1
make[2]: Leaving directory '/home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver'
make[1]: *** [Makefile:149: for-all-target-./receiver/prometheusreceiver] Error 2
make[1]: Leaving directory '/home/jpkroehling/Projects/src/github.com/open-telemetry/opentelemetry-collector-contrib'
make: *** [Makefile:44: unit-tests-with-cover] Error 2
```

**Comment by djaglowski** [comment]
Again here: https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/7847031428?check_suite_focus=true#step:8:1200

**Comment by crobert-1** [comment]
I believe this was fixed by #20889.
